x-pack/plugins/security_solution/public/common/components/sourcerer/use_pick_index_patterns.tsx

```diff
@@ -172,9 +181,47 @@ export const usePickIndexPatterns = ({
     []
   );
 
-  const setIndexPatternsByDataView = (newSelectedDataViewId: string, isAlerts?: boolean) => {
-    setSelectedOptions(getDefaultSelectedOptionsByDataView(newSelectedDataViewId, isAlerts));
-  };
+  const setIndexPatternsByDataView = useCallback(
+    async (newSelectedDataViewId: string, isAlerts?: boolean) => {
```

Comment:
this file is meant to be a hook for manipulating UI elements. Please move the async code over into its own hook in `x-pack/plugins/security_solution/public/common/containers/sourcerer`

Reply:
I think it will have to be a bigger refactor because we only want to fetch the data on selection of the data view and if it is not already in our store. I was wondering if this hook can just be transform to just helpers function.

Reply:
i agree